addons/hr_holidays/tests/test_leave_requests.py

```diff
@@ -884,14 +913,15 @@ def test_archived_allocation(self):
 
             # The holidays count only takes into account the valid allocations at that date
             self._check_holidays_count(
-                self.holidays_type_2.get_employees_days([self.employee_emp_id], date=date(2021, 12, 1))[self.employee_emp_id][self.holidays_type_2.id],
-                ml=10, lt=5, rl=5, vrl=5, vlt=5, closest_allocation=allocation_2021,
+                self.holidays_type_2.get_allocation_data(self.employee_emp, date=date(2021, 12, 1))[self.employee_emp][0][1],
+                ml=10, lt=5, rl=5, vrl=5, vlt=5,
             )
 
-            # Virtual remaining leave is equal to 1 because there is only one day remaining in the allocation based on its validity
-            self._check_holidays_count(
-                self.holidays_type_2.get_employees_days([self.employee_emp_id], date=date(2021, 12, 31))[self.employee_emp_id][self.holidays_type_2.id],
-                ml=10, lt=5, rl=5, vrl=1, vlt=5, closest_allocation=allocation_2022,
+            # Days remaining before the allocation ends is equal to 1 because there is only one day remaining in the allocation based on its validity
+            self.assertEqual(
+                self.holidays_type_2.get_allocation_data(self.employee_emp, date=date(2021, 12, 31))[self.employee_emp][0][1]['closest_allocation_duration'],
+                1,
+                "Only one day should remain before the allocation expires"
```

Comment:
I thought you changed this so that the remaining number of days would be unaffected by the validity period of the allocation. So if you have ten days left, they are ten days even if the allocation expires today?

Reply:
Indeed! I compare the days *before the allocation expires* and not the amount of days remaining :) The actual data of what's remaining is checked just above.